extractors/cds/tools/src/cds/parser/functions.ts

```diff
@@ -0,0 +1,1259 @@
+import { existsSync, readFileSync, statSync } from 'fs';
+import { basename, dirname, join, relative, sep } from 'path';
+
+import { sync } from 'glob';
+
+import {
+  CdsAccessControl,
+  CdsAnnotation,
+  CdsEntity,
+  CdsExposedEntity,
+  CdsImport,
+  CdsParseResult,
+  CdsProperty,
+  CdsService,
+  FileCache,
+  PackageJson,
+} from './types';
+import { cdsExtractorLog } from '../../logging';
+
+// Global file cache to avoid multiple reads of the same file
+const fileCache: FileCache = {
```

Comment:
I don't think we actually need the `FileCache`. It appears to only be used (once unused functions have been deleted) for caching `package.json` files, but I believe those will only be loaded once by the `buildCdsProjectDependencyGraph` function in any case. If we remove it, we can also remove the associated `readFileWithCache` function.